dbms/src/Common/tests/gtest_tiflash_security.cpp

```diff
@@ -49,32 +51,108 @@ TEST(TiFlashSecurityTest, Config)
 cert_path="security/cert.pem"
 key_path="security/key.pem"
 cert_allowed_cn="tidb"
-        )";
-    auto new_config = loadConfigFromString(test);
-    tiflash_config.update(*new_config);
-    ASSERT_EQ((int)tiflash_config.allowedCommonNames().count("tidb"), 1);
+        )");
+        TiFlashSecurityConfig tiflash_config(log);
+        tiflash_config.init(*new_config);
+        ASSERT_TRUE(tiflash_config.hasTlsConfig());
+        ASSERT_EQ(tiflash_config.allowedCommonNames().count("tidb"), 1);
+    }
 
-    test =
-        R"(
+    {
+        auto new_config = loadConfigFromString(R"(
 [security]
 cert_allowed_cn="tidb"
-        )";
-    new_config = loadConfigFromString(test);
-    auto new_tiflash_config = TiFlashSecurityConfig(log);
-    new_tiflash_config.init(*new_config);
-    ASSERT_EQ((int)new_tiflash_config.allowedCommonNames().count("tidb"), 0);
+        )");
+        auto new_tiflash_config = TiFlashSecurityConfig(log);
+        new_tiflash_config.init(*new_config);
+        ASSERT_FALSE(new_tiflash_config.hasTlsConfig());
+        // allowed common names is ignore when tls is not enabled
+        ASSERT_EQ(new_tiflash_config.allowedCommonNames().count("tidb"), 0);
+    }
+}
+
+TEST(TiFlashSecurityTest, EmptyConfig)
+try
+{
+    const auto log = Logger::get();
+
+    for (const auto & c : Strings{
+             // empty strings
+             R"([security]
+ca_path=""
+cert_path=""
+key_path="")",
+             // non-empty strings with space only
+             R"([security]
+ca_path="  "
+cert_path=""
+key_path="")",
+         })
+    {
+        TiFlashSecurityConfig tiflash_config(log);
+        auto new_config = loadConfigFromString(c);
+        tiflash_config.init(*new_config);
+        ASSERT_FALSE(tiflash_config.hasTlsConfig()) << fmt::format("case: {}", c);
+    }
+}
+CATCH
+
+TEST(TiFlashSecurityTest, InvalidConfig)
+try
+{
+    const auto log = Logger::get();
+
+    for (const auto & c : Strings{
+             // only a part of ssl path is set
+             R"([security]
+ca_path="security/ca.pem"
+cert_path=""
+key_path="")",
+             R"([security]
+ca_path=""
+cert_path="security/cert.pem"
+key_path="")",
+             R"([security]
+ca_path=""
+cert_path=""
+key_path="security/key.pem")",
+             R"([security]
+ca_path=""
+cert_path="security/cert.pem"
+key_path="security/key.pem")",
+             // comment out
+             R"([security]
+ca_path="security/ca.pem"
+#cert_path="security/cert.pem"
+key_path="security/key.pem")",
+         })
+    {
+        TiFlashSecurityConfig tiflash_config(log);
+        auto new_config = loadConfigFromString(c);
+        try
+        {
+            tiflash_config.init(*new_config);
+            ASSERT_FALSE(true) << fmt::format("should raise exception, case: {}", c);
+        }
+        catch (Exception & e)
+        {
+            // has_tls remain false when exception raise
+            ASSERT_FALSE(tiflash_config.hasTlsConfig()) << fmt::format("case: {}", c);
```

Comment:
```suggestion
            ASSERT_FALSE(tiflash_config.hasTlsConfig());
```

Reply:
You can use gtest's `SCOPED_TRACE` to automatically trace the current case.

Reply:
Addressed